tests/eos-test-utils.c

```diff
@@ -538,21 +675,50 @@ update_commits (EosTestSubserver *subserver,
   g_hash_table_iter_init (&iter, subserver->ref_to_commit);
   while (g_hash_table_iter_next (&iter, &ref_ptr, &commit_ptr))
     {
+      const gchar *ref = ref_ptr;
       guint commit = GPOINTER_TO_UINT (commit_ptr);
       g_autofree gchar *checksum = NULL;
+      g_autofree gchar *old_checksum = NULL;
 
-      if (!prepare_commit (subserver->repo,
-                           subserver->tree,
-                           commit,
-                           ref_ptr,
-                           subserver->gpg_home,
-                           subserver->keyid,
-                           &checksum,
-                           error))
-        return FALSE;
+      if (commit > 0)
+        {
+          if (!get_current_commit_checksum (subserver->repo,
+                                            ref,
+                                            &old_checksum,
+                                            error))
+            return FALSE;
+
+          if (!prepare_commit (subserver->repo,
+                               subserver->tree,
+                               commit,
+                               ref,
+                               subserver->gpg_home,
+                               subserver->keyid,
+                               &checksum,
+                               error))
+            return FALSE;
+
+          if (!generate_delta_files (subserver->repo,
+                                     old_checksum,
+                                     checksum,
+                                     error))
+            return FALSE;
+        }
+      else
+        {
+          if (!prepare_commit (subserver->repo,
```

Comment:
Nitpick: I’d refactor this so that there’s only one call to `prepare_commit()`, and that the `get_current_commit_checksum()` and `generate_delta_files()` calls are both conditional. Should make it a bit clearer that preparing the commit is the main operation here.